Project/Assets/ML-Agents/Examples/Walker/Scripts/WalkerAgent.cs

```diff
@@ -69,36 +95,35 @@ public override void Initialize()
     /// </summary>
     public void CollectObservationBodyPart(BodyPart bp, VectorSensor sensor)
     {
-        var rb = bp.rb;
+        //GROUND CHECK
         sensor.AddObservation(bp.groundContact.touchingGround ? 1 : 0); // Is this bp touching the ground
-        sensor.AddObservation(rb.velocity);
-        sensor.AddObservation(rb.angularVelocity);
-        var localPosRelToHips = hips.InverseTransformPoint(rb.position);
-        sensor.AddObservation(localPosRelToHips);
+        
+        //Get velocities in the context of our orientation cube's space
+        //Note: You can get these velocities in world space as well but it may not train as well.
+        sensor.AddObservation(m_OrientationCube.transform.InverseTransformDirection(bp.rb.velocity));
+        sensor.AddObservation(m_OrientationCube.transform.InverseTransformDirection(bp.rb.angularVelocity));
+        
+        //Get position relative to hips in the context of our orientation cube's space
+        sensor.AddObservation(m_OrientationCube.transform.InverseTransformDirection(bp.rb.position - hips.position));
 
-        if (bp.rb.transform != hips && bp.rb.transform != handL && bp.rb.transform != handR &&
-            bp.rb.transform != footL && bp.rb.transform != footR && bp.rb.transform != head)
+        if (bp.rb.transform != hips && bp.rb.transform != handL && bp.rb.transform != handR)
         {
-            sensor.AddObservation(bp.currentXNormalizedRot);
-            sensor.AddObservation(bp.currentYNormalizedRot);
-            sensor.AddObservation(bp.currentZNormalizedRot);
+            sensor.AddObservation(bp.rb.transform.localRotation);
             sensor.AddObservation(bp.currentStrength / m_JdController.maxJointForceLimit);
         }
     }
-
+    
```

Comment:
I think formatting should have removed this tab. Why is this here?

Reply:
removed tab